scripts/communications/notification_manager.py

```
@@ -0,0 +1,1382 @@
+                    context.check_hostname = True
+                    context.verify_mode = ssl.CERT_REQUIRED
+                    self.connection.starttls(context=context)
+            
+            # Authenticate
+            self.connection.login(self.config['username'], self.config['password'])
+            self.last_used = datetime.now()
+            
+            logger.info(f"SMTP connection established to {self.config['smtp_server']}:{self.config['smtp_port']}")
+            
+        except smtplib.SMTPAuthenticationError as e:
+            raise SMTPConfigurationError(f"SMTP authentication failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)